lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Utilites to find the relative position of the sun."""
+
+import numpy as np
+import datetime as dt
+
+from improver.utilities.temporal import iris_time_to_datetime
+from improver.utilities.spatial import (
+    lat_lon_determine, transform_grid_to_lat_lon)
+
+
+def solar_declination(day_of_year):
+    """
+    Calculate the Declination for the day of the year.
+
+    Calculation equivalent to the calculation defined in
+    NOAA Earth System Reseach Lab Low Accuracy Equations
+    https://www.esrl.noaa.gov/gmd/grad/solcalc/sollinks.html
+
+    Args:
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
```

Comment:
Can we raise a `ValueError` for number out of these bounds.  (And then test for this in test_solar.py.)

Reply:
I discussed this with Stephen. I think this is an unnecessary test  because of the way we are calculating day_of_year which would only slow down the processing.

Reply:
I'd agree with you if this were part of the plugin.  But as it is, this is a utility that can be accessed by other functions, which may not calculate the day_of_year in the same way.  I think somewhere in these utilities there should be at least a warning if invalid arguments are provided.